mountpoint-s3-client/src/s3_crt_client/put_object.rs

```diff
@@ -109,6 +82,87 @@ impl S3CrtClient {
             state: S3PutObjectRequestState::CreatingMPU(mpu_created),
         })
     }
+
+    pub(super) async fn put_object_single<'a>(
+        &self,
+        bucket: &str,
+        key: &str,
+        params: &PutObjectParams,
+        contents: impl AsRef<[u8]> + Send + 'a,
+    ) -> ObjectClientResult<PutObjectResult, PutObjectError, S3RequestError> {
+        let span = request_span!(self.inner, "put_object_single", bucket, key);
+        let start_time = Instant::now();
+        let response_headers: Arc<Mutex<Option<Headers>>> = Default::default();
+        let slice = contents.as_ref();
+        let content_length = slice.len();
+        let body = {
+            let mut message = self.new_put_request(bucket, key, params)?;
+            message
+                .set_header(&Header::new("Content-Length", content_length.to_string()))
+                .map_err(S3RequestError::construction_failure)?;
+            let body_input_stream =
+                InputStream::new_from_slice(&self.inner.allocator, slice).map_err(S3RequestError::CrtError)?;
+            message.set_body_stream(Some(body_input_stream));
+
+            let options = S3CrtClientInner::new_meta_request_options(message, S3Operation::PutObjectSingle);
+            let response_headers_writer = response_headers.clone();
+            let on_headers = move |headers: &Headers, _: i32| {
+                *response_headers_writer.lock().unwrap() = Some(headers.clone());
+            };
+            self.inner
+                .make_simple_http_request_from_options(options, span, |_| {}, |_| None, on_headers)?
+        };
+
+        body.await?;
+
+        let elapsed = start_time.elapsed();
+        emit_throughput_metric(content_length as u64, elapsed, "put_object");
+
+        Ok(extract_result(&response_headers))
+    }
+
+    fn new_put_request(
+        &self,
+        bucket: &str,
+        key: &str,
+        params: &PutObjectParams,
+    ) -> Result<S3Message<'_>, S3RequestError> {
+        let mut message = self
+            .inner
+            .new_request_template("PUT", bucket)
+            .map_err(S3RequestError::construction_failure)?;
+
+        let key = format!("/{key}");
+        message
+            .set_request_path(&key)
+            .map_err(S3RequestError::construction_failure)?;
+
+        let checksum_config = match params.trailing_checksums {
+            PutObjectTrailingChecksums::Enabled => Some(ChecksumConfig::trailing_crc32c()),
+            PutObjectTrailingChecksums::ReviewOnly => Some(ChecksumConfig::upload_review_crc32c()),
```

Comment:
How does `ReviewOnly` work with put_object single? Is it just no-op?

Reply:
Yes. Not great, but the only alternative I can think of is using a different params type. Not sure if worth it.

Reply:
I think it's fine to use the same type, but we should update the rustdoc on `PutObjectTrailingChecksums` enum.